tests/FSharp.Compiler.ComponentTests/Signatures/ImpliedSignatureHashTests.fs

```diff
@@ -0,0 +1,299 @@
+﻿module FSharp.Compiler.ComponentTests.Signatures.ImpliedSignatureHashTests
+
+open Xunit
+open FSharp.Test
+open FSharp.Test.Compiler
+
+
+
+[<Theory>]
+
+[<InlineDataAttribute("PrivateModuleAdded",
+(*BEFORE*)"""module MyTest
+type MyRecord = {X:string}"""
+(*AFTER*),"""module MyTest
+type MyRecord = {X:string}
+module private PrivateInnerModule = 
+    let private add a b = a + b""")>]
+    
+[<InlineData("Adding a private let binding",
+(*BEFORE*)"""module Foo
+let a b = b - 1
+"""
+(*AFTER*),"""module Foo
+let a b = b - 1
+let private c = 'd'   """)>]
```

Comment:
@nojaf  This is the test case from you, in the ``Hash should be stable for`` section == hash is not modified when the private binding for 'c' was added.